### PR TITLE
Persist presence

### PR DESCRIPTION
**Pull request by mattkrick**
This moves presence from cashay & socket cluster to relay & rethinkdb.
It works by adding a socketId to a list of `connectedSockets` for the user when they connect via websocket.
When the socket disconnects, it removes that socketId.
This is really neat because now you can see if someone is online by querying them via graphql.

NOTE: we'll need to update dokku's `app.json` with a postdeploy hook of `npm run postdeploy`. That way, when we restart the server, the presence will be cleared. 

TEST:
- [ ] log in, go to a team dash. see that your face has a green dot on it in the upper right hand corner
- [ ] in an incognito window, log in with a different team member, see that the dot turns green.
- [ ] copy the incognito url to a 2nd incognito tab, see that the dot stays green for the 2nd user
- [ ] exit the 2nd incognito tab, see the dot stays green on the first window
- [ ] exit the 1st incognito tab, see the dot turns gray on the first window
- [ ] look in the websocket frames on refresh, see that a VERSION_INFO notification comes through with the current app version

### CODE REVIEW

File: src/server/socketHandlers/scConnectionHandler.js [hunk 1]
```diff
@@ -57,40 +51,44 @@ export default function scConnectionHandler(exchange, sharedDataLoader) {
       asyncIterator.return();
       delete socket.subs[opId];
     });
-    socket.on('disconnect', () => {
+    socket.on('disconnect', async () => {
```
Comment (dan-f) [inline src/server/socketHandlers/scConnectionHandler.js]: No reason to use `async` if there isn't an `await` inside the function, and if we're not returning a promise.

File: src/server/utils/postDeploy.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import getRethink from 'server/database/rethinkDriver';
+
+const flushSocketConnections = async () => {
+  const r = getRethink();
+  return r.table('User')
+    .update({
```
Comment (dan-f) [inline src/server/utils/postDeploy.js]: Does this map over every single user in the table?

File: src/universal/modules/teamDashboard/containers/TeamArchive/TeamArchiveRoot.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ const TeamArchiveRoot = ({atmosphere, match, team}) => {
         return (
           <TransitionGroup appear component={null}>
             {error && <ErrorComponent height={'14rem'} error={error} />}
-            {renderProps &&
+            {renderProps && team &&
```
Comment (dan-f) [inline src/universal/modules/teamDashboard/containers/TeamArchive/TeamArchiveRoot.js]: When would `team` be falsy? Seems like a for the `TeamArchiveRoot` this should never be the case.

File: src/server/utils/postDeploy.js (hunk 2 above)
Comment (mattkrick) [reply src/server/utils/postDeploy.js]: yep! when the server restarts there's no graceful shutdown where the socket.disconnect is guaranteed to be called

File: src/universal/modules/teamDashboard/containers/TeamArchive/TeamArchiveRoot.js (hunk 3 above)
Comment (mattkrick) [reply src/universal/modules/teamDashboard/containers/TeamArchive/TeamArchiveRoot.js]: team is fed as a prop, which comes from `TeamRoot` We have 3 choices: wait for TeamRoot to return from the server before we ask for `TeamArchiveRoot`, build `TeamArchiveRoot` to handle a null team, or ignore the data from the parent & request it again inside `TeamArchiveRoot`. Waiting sucks and requesting the same info twice sucks, so I chose the second option. See #1371 for more detail

File: src/server/socketHandlers/scConnectionHandler.js (hunk 1 above)
Comment (mattkrick) [reply src/server/socketHandlers/scConnectionHandler.js]: well the linter definitely failed. nice catch!

**Comment by mattkrick** [comment]
Yeah, we can move it to redis later, i just wanted to keep this scoped small-ish. Move everything to _a_ database, then we can decide which database.
Redis isn't a free lunch though. By default, it'll still persist everything to disk. To wipe it, we'll need to set its config to wipe on restart & then figure out how to trigger that restart to occur in a postdeploy hook.

**Pull request merged**